src/arrayMethodSort.js

```diff
@@ -4,8 +4,24 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let comparator = compareFunction;
+
+    if (typeof comparator !== 'function') {
+      comparator = function (a, b) {
+        return String(a) > String(b) ? 1 : -1;
+      };
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = 0; j < this.length - 1; j++) {
+        if (comparator(this[j], this[j + 1]) > 0) {
+          [this[j], this[j + 1]] = [this[j + 1], this[j]];
+        }
+      }
+    }
+
+    return this;
   };
```

Comment:
Modifying the prototype of built-in JavaScript objects is generally considered bad practice as it can lead to unexpected behavior in other parts of the code that use the same built-in methods. Instead, consider defining a standalone function that takes an array as an argument.